### ISSUE TITLE
Unable to download foreign file from a URL with query parameters

### ISSUE THREAD

**Issue by Melkaz** [nextflow-io/nextflow#2918]
### Expected behavior and actual behavior

Should download a foreign file from a URL with query parameters (contains a '?' character).
It seems that the URL is stripped of everything after the '?' character.

For example, this url: `https://example.com/file?q=param` will be modified to `https://example.com/`

### Steps to reproduce the problem

```nextflow
#!/usr/bin/env nextflow
nextflow.enable.dsl = 2

process Task {
    input:
        path json
    
    output:
        stdout

    script:
        """
        #!/usr/bin/env python3

        import json

        with open('${json}', 'r') as f:
            results = json.load(f)
            print(results['RelatedTopics'][0]['Text'])
        """
}

workflow {
    main:
        def inputs = [
            "json1" : file("https://api.duckduckgo.com/?format=json&q=hello"),
        ]

        // This will strip '?.*' from the URL and download foreign file 'https://api.duckduckgo.com/'
        def json1Channel = Channel.fromPath(inputs.json1)

        // Will crash because the fallback HTML page will be passed to the task instead of the JSON
        Task(json1Channel)
        Task.out.view()
}
```

### Environment 

* Nextflow version: 22.04.3 build 5703
* Java version: 1.8.0_172
* Operating system: Linux
* Bash version: GNU bash, version 4.2.46(2)-release

**Comment by pditommaso** [comment]
Thanks for reporting this issue. I think currently the query parameters are stripped away. We should look into how to add support for them.

**Comment by pditommaso** [comment]
Likely [here](https://github.com/nextflow-io/nextflow/blob/25941c383d77d8db6d3379b3744c40c7f21027c8/plugins/nf-amazon/src/main/nextflow/cloud/aws/util/S3PathFactory.groovy#L22-L22) and [here](https://github.com/nextflow-io/nextflow/blob/25941c383d77d8db6d3379b3744c40c7f21027c8/plugins/nf-azure/src/main/nextflow/cloud/azure/file/AzPathFactory.groovy#L57-L61) and

**Comment by pditommaso** [comment]
Solved by [43cc851](https://github.com/nextflow-io/nextflow/commit/43cc85118fda9dc5a1ff9a7bcd9f68bf440f6afd)
